tests/readdir.phpt

```diff
@@ -3,17 +3,22 @@ Check listing directory content
 --SKIPIF--
 <?php if (!extension_loaded("shadow")) print "skip"; ?>
 --FILE--
-<?php 
+<?php
 require_once('setup.inc');
 
 function dirread($dir) {
-if ($dh = opendir($dir)) {
-	while (($file = readdir($dh)) !== false) {
-    	echo "filename: $file : filetype: " . filetype("$dir/$file") . "\n";
+    $aTmp=array();
+    if (($dh = dir($dir)) == true) {
+        while (($file = $dh->read()) !== false) {
```

Comment:
why did we change readdir to Directory object? We can provide extra test for that but we shouldn't change original logic.